include/boost/geometry/strategy/spherical/envelope_range.hpp

```diff
@@ -0,0 +1,283 @@
+        envelope::spherical_point::apply(*prev, mbr);
+        return;
+    }
+
+    // initialize box with the first segment
+    envelope_strategy.apply(*prev, *it, mbr);
+
+    // consider now the remaining segments in the range (if any)
+    prev = it;
+    ++it;
+    while (it != end)
+    {
+        using point_t = typename boost::range_value<Range>::type;
+        geometry::model::referring_segment<point_t const> const seg(*prev, *it);
+        expand_strategy.apply(mbr, seg);
+        prev = it;
+        ++it;
+    }
+}
+
+
+// Side of pole WRT segment which doesn't contain it.
+template <typename CalculationType = void>
+struct side_of_pole
+{
+    typedef spherical_tag cs_tag;
+
+    template <typename P>
+    static inline int apply(P const& p1, P const& p2, P const& p)
+    {
+        using calc_t = typename promote_floating_point
+            <
+                typename select_calculation_type_alt
+                    <
+                        CalculationType, P
+                    >::type
+            >::type;
+
+        using units_t = typename geometry::detail::cs_angular_units<P>::type;
+        using constants = math::detail::constants_on_spheroid<calc_t, units_t>;
+
+        calc_t const c0 = 0;
+        calc_t const pi = constants::half_period();
+
+        calc_t const lon1 = get<0>(p1);
+        calc_t const lat1 = get<1>(p1);
+        calc_t const lon2 = get<0>(p2);
+        calc_t const lat2 = get<1>(p2);
+        calc_t const lat = get<1>(p);
+
+        calc_t const s_lon_diff = math::longitude_distance_signed<units_t>(lon1, lon2);
+        bool const s_vertical = math::equals(s_lon_diff, c0)
+                             || math::equals(s_lon_diff, pi);
+        // Side of vertical segment is 0 for both poles.
+        if (s_vertical)
+        {
+            return 0;
+        }
+
+        // This strategy shouldn't be called in this case but just in case
+        // check if segment starts at a pole
+        if (math::equals(lat, lat1) || math::equals(lat, lat2))
+        {
+            return 0;
+        }
+
+        // -1 is rhs
+        //  1 is lhs
+        if (lat >= c0) // north pole
+        {
+            return s_lon_diff < c0 ? -1 : 1;
+        }
+        else // south pole
+        {
+            return s_lon_diff > c0 ? -1 : 1;
+        }
+    }
+};
+
+
+template <typename Point, typename Range, typename Strategy>
+inline int point_in_range(Point const& point, Range const& range, Strategy const& strategy)
+{
+    typename Strategy::state_type state;
+
+    auto it = boost::begin(range);
+    auto const end = boost::end(range);
+    for (auto previous = it++ ; it != end ; ++previous, ++it )
+    {
+        if ( ! strategy.apply(point, *previous, *it, state) )
+        {
+            break;
+        }
+    }
+
+    return strategy.result(state);
+}
+
+
+template <typename Ring, typename PoleWithinStrategy>
+inline bool pole_within(bool north_pole, Ring const& ring,
```

Comment:
Alternatively, of having a `bool` parameter and an `if` statement you could directly pass `max_latitude` or `min_latitude` as a function parameter and explicitly set it in the function.